### ISSUE TITLE
Samsung screen zoom causes layout issue with preset tab

### ISSUE THREAD

**Issue by jajajaneeneenee** [MarcusWolschon/osmeditor4android#2960]
## Vespucci Version
v21.1.0

## Download source
Google Play Store

## Device (Manufacturer and Model)
Samsung A33

## Android Version 
15

## Behaviour/Symptoms
I received the update v21.1.0 on August 5, 2025, and today (August 6, 2025) I experienced a nasty surprise during my first mapping tour with v21.1.0: many of my (custom) presets were no longer displayed in the preset overview in the editor. Entire groups were missing, and sometimes only a few presets within groups. And it looks completely arbitrary. I haven't changed anything in these presets/groups and have never had any issues with them, so it's probably not the preset code.

Despite several tests, I couldn't find any systematic or any reason for this.

A preset search, however, finds the missing presets; nothing is missing from the map icon display either. And if you open an existing object with one of these missing presets in the editor, everything is displayed normally.

However, this completely ruins the mapping experience – especially when it comes to creating new objects. I use the preset overview extensively for this (often without preset search). And I've never experienced anything comparable in Vespucci before.

I also had the beta version of 21.1.0 before – without these problems! This really surprises me now with the v21.1.0 final version. Normally it would be more likely to happen the other way around.

Deleting the (custom) presets in the preset list and creating new ones didn't help either. Exactly the same problem, the same presets are missing.

I also tried it with Simonpoole Beautified Presets v5.81.0 (only activated these) – for example, the main groups "Sports" and "Annotation" are missing. It's too time-consuming to determine further details, such as individual missing presets, but I think this should suffice for an analysis.

Here is how it looks like (if a node is selected – I think the groups “Sports” and “Annotation” should be displayed):

![Image](https://github.com/user-attachments/assets/4a85c121-ac3c-42cd-ad2a-8d4aaf647752)

As a first step, it would be helpful for me to know if this is reproducible for you (with Simonpoole Beautified Presets v5.81.0). But maybe you already have an idea where this problem might come from.

I don't think I have a chance of finding out more, and I can't find a workaround. Or should I reinstall Vespucci from scratch (deleting its entire memory, etc.)? I've been reluctant to go through that effort so far. I only cleared the Vespucci app cache once – no change.

Do you need some more information (like a debug info screenshot or something else)?

**Comment by simonpoole** [comment]
Android 15 emulator works as expected, so it seems to be an Samsung specific issue, stock Android doesn't seem to have the exact same function in any case.

**Comment by simonpoole** [comment]
The effect is caused by the zoomed views returning incorrect dimensions for layouting in WrapperLayout. 

Example:
Screenwidth/container view reported as 1080px wide with and without zoom.

Preset item view dimensions without zoom 222px, with zoom 252px.

But in reality the zoomed items are more like 350px. This might be an issue with padding not being correctly reported or similar.

**Comment by jajajaneeneenee** [comment]
Oh, what a strange effect! I've actually never heard of Samsung's screen zoom before and don't remember if I ever adjusted or changed anything there. It was set to level 2 out of 5 (so if the first zoom is 0, then it was zoom 1). And the reported problems occur (of course!) only at THIS zoom level, levels 0, 2, 3, 4 = OK (only text can be cut off in the 4th line – with long preset names – at the zoom levels 3 and 4).

And I was very surprised by your screenshot with 5 elements in a row. I've always only known it with 4 elements, even on my older phone (an Honor 7x), it was like that; I just checked an old screenshot from 2020. And it must have always been like that on the Samsung, too, because I often divided my custom presets into meaningful blocks with `<separator />` lines, always calculating with 4 in a row.

But I'm glad it's only that... I was afraid of something more difficult.

I can now use zoom level 2 (3 out of 5), with 4 elements in a row, and without the text being cut off in the 4th line.

I took another series of screenshots of one of my custom preset groups (a section of it). Something strange happened: after several attempts, the font size or line spacing (or both?) wasn't 100% identical at the same zoom level (and I hadn't changed anything in the "Font Size and Style" system setting; mine is set to level 3 out of 8). Pretty unexpected. Is Samsung playing dice when setting the font properties? See two screenshots of zoom level 3 below for comparison.

If you look closely, there's another strange phenomenon (purely cosmetic): the dark gray background rectangles (squares?) of individual items don't always have the same height, which is probably due to different font sizes (caused by different lengths of preset names). The light gray backgrounds of groups are always the same because the font is always the same font size (short group names). Can't this height be specified in a fixed unit that isn't affected by any padding/margin/font values/changes?

Screenshot zoom level 0 (scaled; 50% of the original size):

![Image](https://github.com/user-attachments/assets/0b8e450d-10ed-4829-b8a6-dd31dde5dd4e)

Screenshot zoom level 1 (scaled; 50% of the original size):

![Image](https://github.com/user-attachments/assets/4bcbe089-5407-4f50-b355-0fe0846a5d53)
(The groups/articles "Carports", "House SC", and "Roofs" are missing here – exactly those that were previously the fifth in the row – so here is the system...!)

Screenshot zoom level 2 (scaled; 50% of the original size):

![Image](https://github.com/user-attachments/assets/21b9b0a2-5bdd-41a6-ab77-742b4282298e)

Screenshot zoom level 3 (scaled; 50% of the original size):

![Image](https://github.com/user-attachments/assets/6dfdf229-9844-471e-b6e7-4f480cbe683d)

Screenshot zoom level 3 (scaled; 50% of the original size) from another screenshot series some minutes before. (The font is not the same – different size, different line spacing, different kerning? And the line breaks also change in some cases. Perhaps Samsung is playing a joke?):

![Image](https://github.com/user-attachments/assets/c6be6924-6712-49c3-8196-7daf069fffa8)

Screenshot zoom level 4 (scaled; 50% of the original size):

![Image](https://github.com/user-attachments/assets/f6cae7b0-3024-4861-a0ba-04a409f9a3dd)

Maybe you can fix it so that no zoom level causes problems anymore – but then it's at least a minor issue. Unless most people out there with Samsungs also use level 1 (and that always causes the problems somewhat randomly; and maybe a different zoom level (or several levels) can cause problems depending on the screen size/resolution/density, I realize that). Not sure if you can make it rock solid for all possible cases.

**Comment by jajajaneeneenee** [comment]
One thing I forgot to mention: as I wrote above, I really didn't have this issue with v21.1.0 Beta (or any previous version either), so I would have definitely noticed. It only first appeared after the update to v21.1.0 final. Perhaps some strange coincidence?
